From 832961f3c4a2d33710bcf0a05fec0d593ff613c3 Mon Sep 17 00:00:00 2001
From: Bernie Innocenti <codewiz@google.com>
Date: Thu, 14 Jun 2018 19:02:47 +0900
Subject: [PATCH 08/32] Don't crash on readPacketFilter() failure

The crash only occurs when verbose wifi logging is enabled.
It interferes with debugging, but doesn't affect regular users.

Test: manually, by turning on wifi debugging
Bug: 110177912
Change-Id: Ie655f86349394bf0e76a724046aeb2049e614ffe
---
 service/java/com/android/server/wifi/WifiVendorHal.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/service/java/com/android/server/wifi/WifiVendorHal.java b/service/java/com/android/server/wifi/WifiVendorHal.java
index 0d73459b5..50fc0e145 100644
--- a/service/java/com/android/server/wifi/WifiVendorHal.java
+++ b/service/java/com/android/server/wifi/WifiVendorHal.java
@@ -201,7 +201,7 @@ public class WifiVendorHal {
 
         mVerboseLog.err("% returns %")
                 .c(niceMethodName(trace, 3))
-                .c(HexDump.dumpHexString(result))
+                .c(result == null ? "(null)" : HexDump.dumpHexString(result))
                 .flush();
 
         return result;
-- 
2.17.1

